pypdf/generic/_files.py

```diff
@@ -25,29 +41,124 @@ def __init__(self, name: str, pdf_object: DictionaryObject) -> None:
+        """
+        Create a new embedded file and add it to the PdfWriter.
+
+        Args:
+            writer: The PdfWriter instance to add the embedded file to.
+            name: The filename to display.
+            content: The data in the file.
+
+        Returns:
+            EmbeddedFile instance for the newly created embedded file.
+        """
+        # Convert string content to bytes if needed
+        if isinstance(content, str):
+            content = content.encode("latin-1")
+
+        # Create the file entry (the actual embedded file stream)
+        file_entry = DecodedStreamObject()
+        file_entry.set_data(content)
+        file_entry.update({NameObject(PA.TYPE): NameObject("/EmbeddedFile")})
+
+        # Create the /EF entry
+        ef_entry = DictionaryObject()
+        ef_entry.update({NameObject("/F"): writer._add_object(file_entry)})
+
+        # Create the filespec dictionary
+        from pypdf.generic import create_string_object  # noqa: PLC0415
+        filespec = DictionaryObject()
+        filespec.update(
+            {
+                NameObject(PA.TYPE): NameObject("/Filespec"),
+                NameObject(FileSpecificationDictionaryEntries.F): create_string_object(name),
+                NameObject(FileSpecificationDictionaryEntries.EF): ef_entry,
+            }
+        )
+
+        # Add to the catalog's names tree
+        if CA.NAMES not in writer._root_object:
+            writer._root_object[NameObject(CA.NAMES)] = writer._add_object(DictionaryObject())
+
+        names_dict = cast(DictionaryObject, writer._root_object[CA.NAMES])
+        if "/EmbeddedFiles" not in names_dict:
+            embedded_files_names_dictionary = DictionaryObject(
+                {NameObject(CA.NAMES): ArrayObject()}
+            )
+            names_dict[NameObject("/EmbeddedFiles")] = writer._add_object(embedded_files_names_dictionary)
+        else:
+            embedded_files_names_dictionary = cast(DictionaryObject, names_dict["/EmbeddedFiles"])
+
+        # Add the name and filespec to the names array
+        names_array = cast(ArrayObject, embedded_files_names_dictionary[CA.NAMES])
+        names_array.extend([create_string_object(name), filespec])
+
+        # Return an EmbeddedFile instance
+        return cls(name=name, pdf_object=filespec)
+
     @property
     def alternative_name(self) -> str | None:
         """Retrieve the alternative name (file specification)."""
         for key in [FileSpecificationDictionaryEntries.UF, FileSpecificationDictionaryEntries.F]:
             # PDF 2.0 reference, table 43:
             #   > A PDF reader shall use the value of the UF key, when present, instead of the F key.
             if key in self.pdf_object:
-                return cast(str, self.pdf_object[key].get_object())
+                value = self.pdf_object[key].get_object()
+                if not is_null_or_none(value):
```

Comment:
These seems to have incomplete coverage? https://app.codecov.io/gh/py-pdf/pypdf/pull/3374/blob/pypdf/generic/_files.py?dropdown=coverage#L117